app/domain/authentication/authn_jwt/restriction_validators/fetch_mandatory_claims.rb

```diff
@@ -0,0 +1,73 @@
+require 'command_class'
+
+module Authentication
+  module AuthnJwt
+    module RestrictionValidators
+      # Fetch the mandatory claims from the JWT authenticator policy which enforce 
+      # definition of annotations keys on JWT hosts 
+      FetchMandatoryClaims = CommandClass.new(
+        dependencies: {
+          resource_class: ::Resource,
+          fetch_required_secrets: ::Conjur::FetchRequiredSecrets.new,
+          parse_mandatory_claims: ::Authentication::AuthnJwt::InputValidation::ParseMandatoryClaims.new,
+          logger: Rails.logger
+        },
+        inputs: %i[authentication_parameters]
+      ) do
+        def call
+          fetch_mandatory_claims
+        end
+
+        private
+
+        def fetch_mandatory_claims
+          @logger.debug(LogMessages::Authentication::AuthnJwt::FetchingMandatoryClaims.new)
+          if !mandatory_claims_resource_exists?
+            @logger.debug(LogMessages::Authentication::AuthnJwt::NotConfiguredMandatoryClaims.new)
+            return empty_mandatory_claims
+          else
+            fetch_mandatory_claims_secret_value
+            parse_mandatory_claims_secret_value
+          end
+
+          @logger.info(LogMessages::Authentication::AuthnJwt::FetchingMandatoryClaims.new(parse_mandatory_claims_secret_value))
+          parse_mandatory_claims_secret_value
+        end
+        
+        def mandatory_claims_resource_exists?
+          return @mandatory_claims_resource_exists unless @mandatory_claims_resource_exists.nil?
+
+          @mandatory_claims_resource_exists ||= !mandatory_claims_resource.nil?
+        end
+
+        def mandatory_claims_resource
+          @mandatory_claims_resource ||= @resource_class[mandatory_claims_resource_id]
+        end
+
+        def mandatory_claims_resource_id
+          @mandatory_claims_resource_id ||= "#{@authentication_parameters.authn_jwt_variable_id_prefix}/#{MANDATORY_CLAIMS_RESOURCE_NAME}"
+        end
+        
+        def empty_mandatory_claims
+          @empty_mandatory_claims ||= []
+        end
+        
+        def fetch_mandatory_claims_secret_value
```

Comment:
What is the purpose of naming it secret? in the future we could use other way to save it, I dont think we should make the variable permanent

Reply:
Correct, but we have the dependency `FetchRequiredSecrets` to fetch the secret object and then i use it to get the value so i called it ...._secret_value